Library/Homebrew/extend/os/mac/search.rb

```diff
@@ -37,8 +37,7 @@ def search_casks(string_or_regex)
         results = cask_tokens.extend(Searchable)
                              .search(string_or_regex)
 
-        cask_names = Cask::Cask.all.map(&:full_name)
-        results += DidYouMean::SpellChecker.new(dictionary: cask_names)
+        results += DidYouMean::SpellChecker.new(dictionary: cask_tokens)
```

Comment:
A quick explanation of why this `cask_names` calculation isn't needed anymore. Essentially, `cask_names` and `cask_tokens` return almost identical results but, of course, `cask_names` is much slower.

### `cask_names`
https://github.com/Homebrew/brew/blob/ed3789a0bb34464a37acd224f196f906f8d73b6c/Library/Homebrew/extend/os/mac/search.rb#L40

For each cask in a tap we call `Cask#full_name` which either returns the token if it's a core tap or the whole path otherwise (`user/repo/token`).

https://github.com/Homebrew/brew/blob/ed3789a0bb34464a37acd224f196f906f8d73b6c/Library/Homebrew/cask/cask.rb#L115-L120

https://github.com/Homebrew/brew/blob/ed3789a0bb34464a37acd224f196f906f8d73b6c/Library/Homebrew/tap.rb#L102

### `cask_tokens`

https://github.com/Homebrew/brew/blob/ed3789a0bb34464a37acd224f196f906f8d73b6c/Library/Homebrew/extend/os/mac/search.rb#L33-L35

As you can see it returns essentially the same result. It removes the `user/repo` part from core taps and otherwise keeps everything the same. That's why we can use it for both.